services/launcher-dev/app/routes.py

```diff
@@ -128,7 +78,7 @@ def compile_and_test():
 
     # We copy the source file from /uploads to a fresh directory in /compilations
     dir_for_compilation = basename
-    path_for_compilations = os.path.join('/compilations', dir_for_compilation)
+    path_for_compilations = os.path.join('/uploads/compilations', dir_for_compilation)
```

Comment:
Actually, if you don't add `/uploads` here, the path for complications in created in the `launcher`, which cannot be accessed by the `complie_and_test` service.